windows/windows_update_install/prepare_msu_file.yml

```diff
@@ -41,26 +60,47 @@
 - name: "Copy the .msu file to local disk of guest OS"
   include_tasks: ../utils/win_execute_cmd.yml
   vars:
+    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
     win_powershell_cmd: >-
       {{ win_nfs_mount_cmd }};
-      Copy-Item -Path {{ msu_file_src_path }} -Include *.msu -Destination {{ msu_dir_path }}  -ErrorAction Stop;
-      net use {{ drive_letter_new }}: /delete
-    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
+      $file = Get-ChildItem -Path {{ msu_nfs_dir }} | where Name -like '{{ msu_file_name_reg }}';
+      if ($file) {Copy-Item -Path $file.FullName -Destination {{ msu_dest_dir }} -ErrorAction Stop;};
+      net use {{ drive_letter_new }}: /delete;
+      Write-Host $file.FullName
 
-- name: "Get the .msu file name"
-  include_tasks: ../utils/win_execute_cmd.yml
-  vars:
-    win_powershell_cmd: >- 
-      Get-ChildItem -Path {{ msu_dir_path }} -Include *.msu -Name -ErrorAction Stop;
+- name: "Check if the .msu file exists on source"
+  ansible.builtin.set_fact:
+    msu_nfs_path_list: "{{ win_powershell_cmd_output.stdout_lines | select('search', '(?i)windows.*\\.msu$') }}"
+  when:
+    - win_powershell_cmd_output.stdout_lines is defined
+    - win_powershell_cmd_output.stdout_lines | length > 0
 
-- name: "Check if the .msu file is copied to {{ msu_dir_path }}"
+- name: "Verify the .msu file of type {{ msu_file_type }} exists on source"
   ansible.builtin.assert:
     that:
-      - win_powershell_cmd_output.stdout_lines is defined
-      - win_powershell_cmd_output.stdout_lines | length != 0
-    fail_msg: "The .msu file is not found in {{ msu_dir_path }} in guest OS."
-    success_msg: "The .msu file is copied to {{ msu_dir_path }} in guest OS."
+      - msu_nfs_path_list | length == 1
+    fail_msg: >-
+      The .msu file of type {{ msu_file_type }} is essential. But it's not found in shared folder: {{ windows_nfs_msu_path }}.
```

Comment:
Here can add a `success_msg` to display found .msu file on source `msu_nfs_path_list[0]`

Reply:
Added. Thanks.